src/Build/Definition/Project.cs

```diff
@@ -1117,32 +1129,161 @@ public List<GlobResult> GetAllGlobs(string itemType)
             return GetAllGlobs(GetItemElementsByType(GetEvaluatedItemElements(), itemType));
         }
 
+        private struct CumulatedRemoveElementData
+        {
+            public ImmutableList<IMSBuildGlob>.Builder Globs { get; set; }
```

Comment:
True, I could encapsulate it and provide CumulateRemoveGlob / GetImmutableSnapshot type of methods. However, since it's a private implementation struct I did not consider it too much (If I could have defined structs inside methods I would have done that). I'll play around with it and see hot it looks encapsulated.